lib/vmdb/logging.rb

```diff
@@ -30,8 +30,7 @@ class LogProxy < Struct.new(:klass, :separator)
     def prefix(location = caller_locations(1, 1))
       location = location.first if location.kind_of?(Array)
       meth = location.base_label
-      meth = meth ? "#{klass}#{separator}#{meth}" : klass
-      "MIQ(#{meth})"
+      prefix_cache[meth] ||= meth ? "MIQ(#{klass}#{separator}#{meth})" : "MIQ(#{klass})"
```

Comment:
I don't think I'm opposed to building the string cache, but I don't think `meth` is specific enough key, is it?

Reply:
I think it is (incoming long explanation...):

Both the instance and class methods of `_log` basically are defined on the class level, and so there is a `@instance_logger` for instance methods and a `@_log` for class methods (clearly less descriptive of a name...) defined just below all of this code [here](https://github.com/NickLaMuro/manageiq/blob/cf040e3f/lib/vmdb/logging.rb#L43-L59). 
 Of what we pass in to it on instantiation, we pass in the `klass` and `separator` as effectively constants for that instance.

When generating the prefix string, it takes `klass`, `separator`, and `meth`, and of those three the only thing that is derived in this method is `meth`, which should work just fine in the cache since it is the only unique thing per instance.

* * *

The only concern I have is that this is a shared class variable, and there might be some weirdness with threads, but I don't think that should be an issue since they should be basically writing to the same values to the cache regardless... but I might be missing something.

Also, this will now not be garbage collected, so this might add some bloat to the application which we are already guilty of having at this point.